app/src/lib/stores/accounts-store.ts

```diff
@@ -106,10 +96,16 @@ export class AccountsStore extends TypedBaseStore<ReadonlyArray<Account>> {
       return null
     }
 
-    this.accounts = [...this.accounts, updated]
+    const accountsByEndpoint = this.accounts.reduce(
```

Comment:
This ensures that we only ever have one account per endpoint. While it's been possible to add more in the past through a bug in the welcome wizard the token for the account is stored using the endpoint as key so it's never been possible to actually have more than one valid token per endpoint.